### PR TITLE
drivers: i2c: stm32: cleanup build issue if no I2C ports are configured

### PR DESCRIPTION
**Pull request by galak**
We can enable the driver, but not actually have any I2C ports enabled so
we will get build issues in this case.  Put ifdef protection around the
code so that we only build it if one of the ports is enabled.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Review by ydamigos — COMMENTED** [review]
What about the case that someone forgets to enable a I2C port? He will get no error and he will spend more time debugging the issue. If someone does not need the driver, he should disable it.

**Comment by galak** [comment]
Well the port enablement should eventually be driven by DT, so we really shouldn't have that issue.

File: drivers/i2c/i2c_ll_stm32.c [hunk 1]
```diff
@@ -17,6 +17,9 @@
 #define SYS_LOG_LEVEL CONFIG_SYS_LOG_I2C_LEVEL
 #include <logging/sys_log.h>
 
+#if defined(CONFIG_I2C_0) || defined(CONFIG_I2C_1) || \
```
Comment (galak) [inline drivers/i2c/i2c_ll_stm32.c]: Thanks, fixed.

**Comment by mnkp** [comment]
@ydamigos made a good point. The change hides the problem and makes it more difficult to debug the issue. The compiler warning in case no ports are enabled is very much in place and helpful. We would better make it more not less explicit. E.g. by having something like the following check at the top of the file:
```
#if !(defined(CONFIG_I2C_1) || defined(CONFIG_I2C_2) || defined(CONFIG_I2C_3))
#error "No I2C port is enabled"
#endif
```
as en error or warning. If we don't want the warning it's better to leave the code the way it is now / abandon this PR.

**Comment by galak** [comment]
The compile error breaks testsuite runs.  As I mentioned once we put the DT stuff in, this will not mater, as we'll use the DT configure a port being enabled or not.

**Comment by mnkp** [comment]
But does it not mean that we configured the testcase incorrectly? So we should rather fix the testcase.

**Comment by galak** [comment]
Yeah, so I looked into this and worked up a different solution.  We should fix the testcases, but will look at that in the future.

**Pull request merged**